experiment/migrate_testgrid_tabs.py

```diff
@@ -0,0 +1,197 @@
+        confFile.truncate()
+
+
+def move_tab(dashboard, dashTab, prow_dir):
+    """ moveTab determines if the dashboard tab matches a prow job in prow_dir.
+    If it does, it modifies the prow job file to include the necessary annotations
+    It also deletes from the config the dashboard tab, so there are no duplicates.
+    """
+    dashboardName = dashboard["name"]
+    prowJobName = dashTab["test_group_name"]
+    prowJobFileName = find_prow_job(prowJobName, prow_dir)
+    if prowJobFileName == "":
+        return
+
+    # Matching file found; patch and delete
+    print("Patching {0} in {1}".format(prowJobName, prowJobFileName))
+
+    with open(prowJobFileName, "r") as prowJobFile:
+        prowConfig = ruamel.yaml.load(prowJobFile,
+                                      Loader=ruamel.yaml.RoundTripLoader,
+                                      preserve_quotes=True)
+
+    # For each presubmits, postsubmits, periodic:
+    # presubmits -> <any repository> -> [{name: prowjob}]
+    if "presubmits" in prowConfig:
+        for repo, jobs in prowConfig["presubmits"].items():
+            for (i, job) in enumerate(jobs):
+                if prowJobName == job["name"]:
+                    prowConfig["presubmits"][repo][i] = patch_prow_job(job,
+                                                                       dashTab,
+                                                                       dashboardName)
+    # postsubmits -> <any repository> -> [{name: prowjob}]
+    if "postsubmits" in prowConfig:
+        for repo, jobs in prowConfig["postsubmits"].items():
+            for (i, job) in enumerate(jobs):
+                if prowJobName == job["name"]:
+                    prowConfig["postsubmits"][repo][i] = patch_prow_job(job,
+                                                                        dashTab,
+                                                                        dashboardName)
+    # periodics -> [{name: prowjob}]
+    if "periodics" in prowConfig:
+        for i in range(0, len(prowConfig["periodics"])):
+            if prowJobName == prowConfig["periodics"][i]["name"]:
+                prowConfig["periodics"][i] = patch_prow_job(prowConfig["periodics"][i],
+                                                            dashTab,
+                                                            dashboardName)
+    # Dump ProwConfig to prowJobFile
+    with open(prowJobFileName, "w") as prowJobFile:
+        ruamel.yaml.dump(prowConfig, prowJobFile, Dumper=ruamel.yaml.RoundTripDumper)
+        prowJobFile.truncate()
+
+    # delete tab
+    dashboard["dashboard_tab"].remove(dashTab)
+
+
+def assert_tab_keys(tab):
+    """Assert that the dashboard tab doesn't contain anything not on the
+    following whitelist:
+    name, description, test_group_name, alert_options->alert_mail_to_addresses
+    """
+    failingKeys = [st for st in tab.keys() if st != "name"
```

Comment:
List comprehensions should probably be a one-liner. I think you can also shorten to:

whitelist = ["name", "description", "test_group_name", "alert_options"]
if any([st for st in tab.keys() if st not in whitelist]):
  return False